art_internal_impl.hpp

```diff
@@ -662,6 +756,200 @@ class basic_inode_impl : public ArtPolicy::header_type {
     // LCOV_EXCL_STOP
   }
 
+  // Return the child node at the specified child index.
+  //
+  // Note: For N48, the child_index is the index into the
+  // child_indices[].  For all other node types, it is a direct index
+  // into the children[].  This method hides this distinction.
+  [[nodiscard]] constexpr node_ptr get_child(node_type type, std::uint8_t child_index) noexcept { // TODO Make const?
```

Comment:
Please make const

Reply:
This one is a can of worms.  I chased it for a while and then backed things out and just left the TODO there.  I can remove the TODO.

Reply:
Ack, but please keep the TODO, this is something to look into after merging this